src/pal/src/include/pal/stackstring.hpp

```diff
@@ -114,6 +140,16 @@ class StackString
         return Set(s.m_buffer, s.m_count);
     }
 
+    //If takeOwnerShip set to TRUE- This object will take over the lifetime
+    //management of the buffer
+    void SetLiteral(const T * buffer, SIZE_T count, BOOL takeOwnerShip = FALSE)
```

Comment:
I don't like this change. It complicates the code in the StackString a lot and you are using the SetLiteral at one place only (not counting the Append where you can just implement the version for the T* parameter as the base one and call it from the version for the StackString& parameter).